### ISSUE TITLE
Feature request: get notified when a contact is added/updated/deleted

### ISSUE THREAD

**Issue by KishanBagaria** [codebytere/node-mac-contacts#7]
So I ran into the issue when a contact was added/updated and my app didn't reflect the changes. I can poll `getAllContacts()` every few seconds but it's not ideal.

[Found](https://stackoverflow.com/questions/38331920/cncontactstoredidchangenotification-not-fired-on-mac-os-x) some Swift code that implements this:

```swift
fileprivate var CNContactStore: CNContactStore? = nil

func addObserver() {
    contactStore = CNContactStore()
    NSNotificationCenter.default.addObserver(self, selector: #selector(addressBookDidChange(notification:)), name: NSNotification.Name.CNContactStoreDidChange, object: nil)
}
```

Would you able to add in support for a method that calls a callback with newly updated/added contacts? Would also be a good idea to return [the ID](https://developer.apple.com/documentation/contacts/cncontact/1403103-identifier) of the contact with `getAllContacts()` so that it's possible to update the existing in-memory contact.

Very grateful for your work! 😄

**Comment by codebytere** [comment]
Oh interesting! No promises for how quickly i can get to it but I'll try to poke into it by the end of next week. You're also welcome to take a stab at a PR if you like :)

**Comment by KishanBagaria** [comment]
No problem! Would have attempted a PR but I believe this goes into experienced with NAPI & Obj-C territory :)

**Comment by codebytere** [comment]
@KishanBagaria Objective-C makes this somewhat awful frankly lol but this commit should at least tell you when the `CNContactStore` changed: a15c382 

I'll try to publish a new version at some point soon but you can also just update your fork and poke at it in the meantime

**Comment by codebytere** [comment]
Found a persistence issue so i'll need to work on it a bit more before publishing

**Comment by codebytere** [comment]
Event listening functionality published as [`v1.2.0`](https://github.com/codebytere/node-mac-contacts/releases/tag/v1.2.0)

**Comment by KishanBagaria** [comment]
Thank you! :)
